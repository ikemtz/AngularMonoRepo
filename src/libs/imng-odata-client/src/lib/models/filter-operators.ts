import { IdType } from 'imng-nrsrx-client-utils';
import {
  serializeArrayFilter,
  serializeFunctionFilter,
  serializeSimpleFilter,
} from './filter-serializers';

/**
 * Represents the list of supported simple filter operators.
 */
export class FilterOperators {
  /**
   * The `eq` operator.
   */
  public static readonly EqualTo = (field: string, value: IdType): string =>
    serializeSimpleFilter(field, 'eq', value);
  /**
   * The `gt` operator.
   */
  public static readonly GreaterThan = (field: string, value: IdType): string =>
    serializeSimpleFilter(field, 'gt', value);
  /**
   * The `ge` operator.
   */
  public static readonly GreaterThanOrEqual = (
    field: string,
    value: IdType,
  ): string => serializeSimpleFilter(field, 'ge', value);
  /**
   * The `lt` operator.
   */
  public static readonly LessThan = (field: string, value: IdType): string =>
    serializeSimpleFilter(field, 'lt', value);
  /**
   * The `le` operator.
   */
  public static readonly LessThanOrEqual = (
    field: string,
    value: IdType,
  ): string => serializeSimpleFilter(field, 'le', value);
  /**
   * The `ne` operator.
   */
  public static readonly NotEqualTo = (field: string, value: IdType): string =>
    serializeSimpleFilter(field, 'ne', value);
  /**
   * The `in` operator.
   */
  public static readonly In = (field: string, values: IdType[]): string =>
    serializeArrayFilter(field, 'in', values);
  /**
   * The `in` operator.
   */
  public static readonly NotIn = (field: string, values: IdType[]): string =>
    serializeArrayFilter(field, 'not in', values);
  /**
   * The `isnotnull` operator.
   */

  public static readonly IsNotNull = (field: string): string =>
    `${field} ne null`;
  /**
   * The `isnull` operator.
   */

  public static readonly IsNull = (field: string): string => `${field} eq null`;
  /**
   * The `contains` operator.
   */
  public static readonly Contains = (field: string, value: IdType): string =>
    serializeFunctionFilter(field, 'contains', value);
  /**
   * The `doesnotcontain` operator.
   */
  public static readonly DoesNotContain = (
    field: string,
    value: IdType,
  ): string => serializeFunctionFilter(field, 'not contains', value);
  /**
   * The `endswith` operator.
   */
  public static readonly EndsWith = (field: string, value: IdType): string =>
    serializeFunctionFilter(field, 'endswith', value);
  /**
   * The `doesnotendwith` operator.
   */
  public static readonly DoesNotEndWith = (
    field: string,
    value: IdType,
  ): string => serializeFunctionFilter(field, 'not endswith', value);
  /**
   * The `startswith` operator.
   */
  public static readonly StartsWith = (field: string, value: IdType): string =>
    serializeFunctionFilter(field, 'startswith', value);
  /**
   * The `doesnotstartwith` operator.
   */
  public static readonly DoesNotStartWith = (
    field: string,
    value: IdType,
  ): string => serializeFunctionFilter(field, 'not startswith', value);
  /**
   * The `isempty` operator.
   */

  public static readonly IsEmpty = (field: string): string =>
    serializeSimpleFilter(field, 'eq', '');
  /**
   * The `isnotempty` operator.
   */

  public static readonly IsNotEmpty = (field: string): string =>
    serializeSimpleFilter(field, 'ne', '');
}
