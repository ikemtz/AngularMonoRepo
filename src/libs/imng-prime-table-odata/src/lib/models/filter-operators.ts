import { FilterOperators, IFilterOperators } from 'imng-odata-client';

export const primeFilterOperators: IFilterOperators = {
  contains: FilterOperators.contains,
  endsWith: FilterOperators.endsWith,
  equals: FilterOperators.equals,
  dateIs: FilterOperators.equals,
  gt: FilterOperators.greaterThan,
  dateAfter: FilterOperators.greaterThan,
  gte: FilterOperators.greaterThanOrEquals,
  in: FilterOperators.in,
  isEmpty: FilterOperators.isEmpty,
  isNull: FilterOperators.isNull,
  lt: FilterOperators.lessThan,
  dateBefore: FilterOperators.lessThan,
  lte: FilterOperators.lessThan,
  notContains: FilterOperators.notContains,
  notEmpty: FilterOperators.notEmpty,
  notEndsWith: FilterOperators.notEndsWith,
  notEquals: FilterOperators.notEquals,
  dateIsNot: FilterOperators.notEquals,
  notIn: FilterOperators.notIn,
  notNull: FilterOperators.notNull,
  notStartsWith: FilterOperators.notStartsWith,
  startsWith: FilterOperators.startsWith,
};
