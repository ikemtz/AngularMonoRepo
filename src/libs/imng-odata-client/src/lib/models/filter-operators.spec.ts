import { FilterOperators } from './filter-operators';

describe('FilterOperators', () => {
  it('should support contains', () => {
    const result = FilterOperators.contains.toODataString('xyz', 123 as never);
    expect(result).toMatchSnapshot();
  });
  it('should support endsWith', () => {
    const result = FilterOperators.endsWith.toODataString('xyz', 123 as never);
    expect(result).toMatchSnapshot();
  });
  it('should support equals', () => {
    const result = FilterOperators.equals.toODataString('xyz', 123 as never);
    expect(result).toMatchSnapshot();
  });
  it('should support greaterThan', () => {
    const result = FilterOperators.greaterThan.toODataString(
      'xyz',
      123 as never,
    );
    expect(result).toMatchSnapshot();
  });
  it('should support greaterThanOrEquals', () => {
    const result = FilterOperators.greaterThanOrEquals.toODataString(
      'xyz',
      'abc' as never,
    );
    expect(result).toMatchSnapshot();
  });
  it('should support in', () => {
    const result = FilterOperators.in.toODataString('xyz', [
      'abc',
      123,
    ] as never);
    expect(result).toMatchSnapshot();
  });
  it('should support notIn', () => {
    const result = FilterOperators.notIn.toODataString('xyz', [
      'abc',
      123,
    ] as never);
    expect(result).toMatchSnapshot();
  });
  it('should support isEmpty', () => {
    const result = FilterOperators.isEmpty.toODataString('xyz');
    expect(result).toMatchSnapshot();
  });
  it('should support isNull', () => {
    const result = FilterOperators.isNull.toODataString('xyz');
    expect(result).toMatchSnapshot();
  });
  it('should support lessThan', () => {
    const result = FilterOperators.lessThan.toODataString('xyz', 123 as never);
    expect(result).toMatchSnapshot();
  });
  it('should support lessThanOrEquals', () => {
    const result = FilterOperators.lessThanOrEquals.toODataString(
      'xyz',
      123 as never,
    );
    expect(result).toMatchSnapshot();
  });
  it('should support notContains', () => {
    const result = FilterOperators.notContains.toODataString(
      'xyz',
      123 as never,
    );
    expect(result).toMatchSnapshot();
  });
  it('should support notEmpty', () => {
    const result = FilterOperators.notEmpty.toODataString('xyz', 123 as never);
    expect(result).toMatchSnapshot();
  });
  it('should support notEndsWith', () => {
    const result = FilterOperators.notEndsWith.toODataString(
      'xyz',
      123 as never,
    );
    expect(result).toMatchSnapshot();
  });
  it('should support notEquals', () => {
    const result = FilterOperators.notEquals.toODataString('xyz', 123 as never);
    expect(result).toMatchSnapshot();
  });
  it('should support notNull', () => {
    const result = FilterOperators.notNull.toODataString('xyz');
    expect(result).toMatchSnapshot();
  });
  it('should support notStartsWith', () => {
    const result = FilterOperators.notStartsWith.toODataString(
      'xyz',
      '123' as never,
    );
    expect(result).toMatchSnapshot();
  });
  it('should support startsWith', () => {
    const result = FilterOperators.startsWith.toODataString(
      'xyz',
      '123' as never,
    );
    expect(result).toMatchSnapshot();
  });
});
