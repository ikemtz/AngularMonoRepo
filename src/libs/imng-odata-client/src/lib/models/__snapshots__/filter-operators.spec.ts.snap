// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterOperators should support contains 1`] = `"contains(xyz,123)"`;

exports[`FilterOperators should support endsWith 1`] = `"endswith(xyz,123)"`;

exports[`FilterOperators should support equals 1`] = `"xyz+eq+123"`;

exports[`FilterOperators should support greaterThan 1`] = `"xyz+gt+123"`;

exports[`FilterOperators should support greaterThanOrEquals 1`] = `"xyz+ge+'abc'"`;

exports[`FilterOperators should support in 1`] = `"xyz+in+('abc',123)"`;

exports[`FilterOperators should support isEmpty 1`] = `"xyz+eq+''"`;

exports[`FilterOperators should support isNull 1`] = `"xyz+eq+null"`;

exports[`FilterOperators should support lessThan 1`] = `"xyz+lt+123"`;

exports[`FilterOperators should support lessThanOrEquals 1`] = `"xyz+le+123"`;

exports[`FilterOperators should support notContains 1`] = `"not+contains(xyz,123)"`;

exports[`FilterOperators should support notEmpty 1`] = `"xyz+ne+''"`;

exports[`FilterOperators should support notEndsWith 1`] = `"not+endswith(xyz,123)"`;

exports[`FilterOperators should support notEquals 1`] = `"xyz+ne+123"`;

exports[`FilterOperators should support notIn 1`] = `"xyz+not+in+('abc',123)"`;

exports[`FilterOperators should support notNull 1`] = `"xyz+ne+null"`;

exports[`FilterOperators should support notStartsWith 1`] = `"not+startswith(xyz,'123')"`;

exports[`FilterOperators should support startsWith 1`] = `"startswith(xyz,'123')"`;
